compiler/rustc_const_eval/src/interpret/intrinsics.rs

```diff
@@ -106,6 +106,13 @@ pub(crate) fn eval_nullary_intrinsic<'tcx>(
             | ty::Tuple(_)
             | ty::Error(_) => ConstValue::from_target_usize(0u64, &tcx),
         },
+        sym::option_some_offset => {
+            let layout = tcx.layout_of(param_env.and(tp_ty)).map_err(|e| err_inval!(Layout(e)))?;
+            let Variants::Multiple { variants, .. } = layout.layout.variants() else { bug!() };
+            let Some(variant) = variants.iter().last() else { bug!() };
+            let FieldsShape::Arbitrary { offsets, .. } = &variant.fields else { bug!() };
+            ConstValue::from_target_usize(offsets[0].bytes(), &tcx)
```

Comment:
I think we should avoid duplicating this code.

Reply:
Ok, where to put it then?